### DISCUSSION TITLE
Integrating with SurrealDB

### DISCUSSION THREAD

**Discussion by unidesigner** [directus/directus#16416] [Feature Requests]
There's a new and interesting new database project called SurrealDB with a SQL-style query language. I was wondering in what way this database could be supported by Directus? For instance, would it be possible to create collections in Directus that could be reused?

**Comment by rijkvanzanten** [comment]
The fact that it uses "SQL style" instead of SQL makes it non-trivial to support, so we'll have to base it on community interest

**Comment by unidesigner** [comment]
How difficult would it be to introduce an SQL-only interface layer between Directus and SurrealDB to simplify integration with Directus, i.e. just map a subset of the SurrealDB interface? A first step may just include supporting CRUD in Directus for schemafull SurrealDB tables which would already be incredibly useful. Is there already something similar in Directus' codebase to evaluate implementation cost?

**Reply by rijkvanzanten** [reply]
In the current way the Directus API is built, we'd need to land support for SurrealDB into [Knex](https://knexjs.org/) first. Once that's completed, support in Directus becomes doable!

**Reply by unidesigner** [reply]
Interestingly, someone asked about SurrealDB [support for Strapi in Knex](https://github.com/knex/documentation/issues/471). For my use case, I may be able to implement a synchronization directly at the database level, i.e. to update the records of a Postgres table based on changes in SurrealDB.

**Reply by rijkvanzanten** [reply]
Strapi also uses Knex as their data abstraction under the hood, so that'd be a win win 🙂

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
